.github/workflows/cicd.yml

```diff
@@ -0,0 +1,238 @@
+        run: echo "${STRATEGY_CONTEXT}";
+
+      - name: Print Matrix Context
+        env:
+          MATRIX_CONTEXT: ${{ toJson(matrix) }}
+        run: echo "${MATRIX_CONTEXT}";
+
+
+  flake8:
+
+    name: Flake8 (linter)
+
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+
+    steps:
+
+      - name: Checkout source code
+        uses: actions/checkout@v2
+
+      - name: Install Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: "3.9"
+
+      - name: Install Tox
+        run: pip install tox;
+
+      - name: Run Flake8
+        run: tox -e flake8;
+
+
+  black:
+
+    name: Black (linter)
+
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+
+    steps:
+
+      - name: Checkout source code
+        uses: actions/checkout@v2
+
+      - name: Install Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: "3.9"
+
+      - name: Install Tox
+        run: pip install tox;
+
+      - name: Run Black
+        run: tox -e black;
+
+
+  mypy:
+    name: Mypy (static type checker)
+
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+
+    steps:
+
+      - name: Checkout source code
+        uses: actions/checkout@v2
+
+      - name: Install Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: "3.9"
+
+      - name: Install Tox
+        run: pip install tox;
+
+      - name: Run Mypy
+        run: tox -e mypy;
+
+
+  docs:
+
+    name: Build documentation
+
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+
+    steps:
+
+      - name: Checkout source code
+        uses: actions/checkout@v2
+
+      - name: Install Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: "3.9"
+
+      - name: Install Tox
+        run: pip install tox;
+
+      - name: Build documentation
+        run: tox -e docs;
+
+
+  packaging:
+    name: Packaging
+
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+
+    steps:
+
+      - name: Checkout source code
+        uses: actions/checkout@v2
+
+      - name: Install Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: "3.9"
+
+      - name: Install Tox
+        run: pip install tox;
+
+      - name: Check packaging
+        run: tox -e packaging;
+
+
+  unit:
+    name: Unit Tests using Python ${{ matrix.python }} on Ubuntu
+
+    needs: [flake8, black, mypy, docs, packaging]
+
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    strategy:
+      matrix:
+        python: ["2.7", "3.5", "3.6", "3.7", "3.8", "3.9", "pypy2", "pypy3"]
+
+    steps:
+
+      - name: Checkout source code
+        uses: actions/checkout@v2
+
+      - name: Install Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Install Tox
+        run: pip install tox;
+
+      - name: Run unit tests
+        shell: bash
+        # This hairy shell code is here to map the Python versions
```

Comment:
You may appreciate how I did this in treq:

https://github.com/twisted/treq/blob/349c44e3ef5f52e635e56bb54727af63efdbeafc/.github/workflows/ci.yaml#L95-L110

Reply:
OMG yeah I like that better. Will revisit when I get some more cycles for that.